File: src/backend/app/schemas/shop.py
from pydantic import BaseModel
from typing import Any


class SalesJsonDict(BaseModel):
    goods_type: str
    days: int
    size: float

class ShopJsonData(BaseModel):
    users_per_day: int
    scene: list[list[str | None]]
    events: dict[str, dict]
    sales: list[SalesJsonDict]

class ShopCreate(BaseModel):
    name: str
    is_generated: bool
    open_at: str
    close_at: str
    data: ShopJsonData

File: src/frontend/src/entities/forms/shop.ts
import { Cashbox, Shelf, StartPoint } from './scene';

export interface NewSalesShopValue {
  goods_type: string;
  size: number;
}

export interface NewShopForm {
  name: string;
  user_per_day: number;
  open_at: string;
  close_at: string;
  events: NewSalesShopValue[];
  sales: NewSalesShopValue[];
  cashboxes: Cashbox[];
  shelves: Shelf[];
  startPoints?: StartPoint;
}
